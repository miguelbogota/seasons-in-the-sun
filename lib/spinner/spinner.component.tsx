import { spinnerStyles } from './spinner.css';

export function Spinner() {
  return (
    <svg className={spinnerStyles.spinner} viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
      <circle className={spinnerStyles.circle} cx="50" cy="50" r="45" />
    </svg>
  );
}
