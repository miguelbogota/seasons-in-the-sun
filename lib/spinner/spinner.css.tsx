import { keyframes, style, theme } from '@app/theme';

const rotate = keyframes({
  '100%': {
    transform: 'rotate(360deg)',
  },
});

const circleAnimation = keyframes({
  '0%, 25%': {
    strokeDashoffset: 280,
    transform: 'rotate(0)',
  },
  '50%, 75%': {
    strokeDashoffset: 75,
    transform: 'rotate(45deg)',
  },
  '100%': {
    strokeDashoffset: 280,
    transform: 'rotate(360deg)',
  },
});

const spinner = style({
  animationName: rotate,
  animationDuration: '2s',
  animationTimingFunction: 'linear',
  animationIterationCount: 'infinite',
  width: '1.75rem',
  overflow: 'visible',
});

const circle = style({
  selectors: {
    [`${spinner} &`]: {
      animationName: circleAnimation,
      animationDuration: '1.4s',
      animationTimingFunction: 'ease-in-out',
      animationIterationCount: 'infinite',
      animationFillMode: 'both',
      display: 'block',
      fill: 'transparent',
      stroke: theme.palette('text.secondary'),
      strokeLinecap: 'round',
      strokeDasharray: 283,
      strokeDashoffset: 280,
      strokeWidth: '0.8rem',
      transformOrigin: '50% 50%',
    },
  },
});

export const spinnerStyles = {
  spinner,
  circle,
};
