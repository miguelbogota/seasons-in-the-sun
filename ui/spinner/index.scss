@use './spinner' as spinner;

@mixin apply {
  @include spinner.apply;
}
