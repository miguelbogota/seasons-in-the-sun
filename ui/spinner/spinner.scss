@mixin apply {
  svg.spinner {
    animation: 2s linear infinite rotate;
    width: 1.75rem;
    overflow: visible;

    @keyframes rotate {
      100% {
        transform: rotate(360deg);
      }
    }

    circle {
      animation: 1.4s ease-in-out infinite both circle-animation;
      display: block;
      fill: transparent;
      stroke: var(--text-secondary);
      stroke-linecap: round;
      stroke-dasharray: 283;
      stroke-dashoffset: 280;
      stroke-width: 0.8rem;
      transform-origin: 50% 50%;
    }

    @keyframes circle-animation {
      0%,
      25% {
        stroke-dashoffset: 280;
        transform: rotate(0);
      }

      50%,
      75% {
        stroke-dashoffset: 75;
        transform: rotate(45deg);
      }

      100% {
        stroke-dashoffset: 280;
        transform: rotate(360deg);
      }
    }
  }
}
