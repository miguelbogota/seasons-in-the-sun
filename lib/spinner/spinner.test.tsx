import { render } from '@testing-library/react';

import { Spinner } from './spinner.component';

describe('Lib / LoadingSpinner', () => {
  it('should match snapshot', () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });
});
