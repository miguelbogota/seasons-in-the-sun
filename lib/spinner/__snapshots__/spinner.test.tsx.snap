// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Lib / LoadingSpinner > should match snapshot 1`] = `
<div>
  <svg
    class="spinner__1pyelcq2"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      class="spinner__1pyelcq3"
      cx="50"
      cy="50"
      r="45"
    />
  </svg>
</div>
`;
